src/lib/vasi-sync/src/lazy_lock.rs

```diff
@@ -112,9 +136,26 @@ where
                     }
                 }
                 InitState::Initializing => {
-                    // Another thread is initializing. Spin.
-                    // TODO: block?
-                    rustix::process::sched_yield();
+                    // `Acquire` ordering on both paths to potentially pair with initialization `Release`.
+                    state = match self.init_state.compare_exchange(
+                        InitState::Initializing.into(),
+                        InitState::InitializingWithSleepers.into(),
+                        atomic::Ordering::Acquire,
+                        atomic::Ordering::Acquire,
```

Comment:
I don't think we need "acquire" in the success case, since there was no "release" to pair with if the state is still "initializing".